src/Strategy/Go/GoListPackages.hs

```diff
@@ -0,0 +1,240 @@
+      | standard = (HashSet.insert importPath stdLibPaths, otherPackages)
+      | otherwise = (stdLibPaths, HashMap.insert importPath (removeIgnoredPackages g) otherPackages)
+
+    -- "C" is a special package for using Go's FFI.
+    -- Ignore it because trying to look it up as a package import path later will fail.
+    ignoredPackages :: HashSet.HashSet ImportPath
+    ignoredPackages = HashSet.insert (ImportPath "C") stdLibImportPaths
+
+    removeIgnoredPackages :: GoPackage -> GoPackage
+    removeIgnoredPackages g@GoPackage{packageDeps = pDeps} =
+      g{packageDeps = filter (\i -> not $ i `HashSet.member` ignoredPackages) pDeps}
+
+    graphEdges :: (Has Diagnostics sig m, Has (Grapher Dependency) sig m) => GoPackage -> m ()
+    graphEdges GoPackage{importPath, packageDeps, listError} = do
+      maybe (pure ()) warn listError
+      currModule@GoModule{isMainModule, indirect} <- importToModule importPath
+      unless (isMainModule || isPathDep currModule) $ do
+        let currDep :: Dependency
+            currDep = modToDep currModule
+
+            addChildEdge :: (Has Diagnostics sig m, Has (Grapher Dependency) sig m) => Dependency -> m ()
+            addChildEdge childDep = do
+              when (childDep /= currDep) $
+                edge currDep childDep
+
+            makeEdge :: (Has (Grapher Dependency) sig m, Has Diagnostics sig m) => ImportPath -> m ()
+            makeEdge = importToModule >=> pure . modToDep >=> addChildEdge
+        traverse_ makeEdge packageDeps
+        if indirect
+          then deep currDep
+          else direct currDep
+
+    -- Look up module info for an import path, performing replacements if necessary.
+    importToModule :: Has Diagnostics sig m => ImportPath -> m GoModule
+    importToModule impPath = do
+      -- Return a stand-in value for missing import paths so it's visible to users? even if unscannable.
+      Diagnostics.fromMaybe (MissingModuleErr impPath) $ do
+        GoPackage{moduleInfo = modInfo} <- HashMap.lookup impPath pkgsNoStdLibImports
+        (modInfo >>= replacement) <|> modInfo
+
+newtype MissingModuleErr = MissingModuleErr ImportPath
+  deriving (Eq, Show)
+
+instance ToDiagnostic MissingModuleErr where
+  renderDiagnostic (MissingModuleErr (ImportPath i)) = pretty $ "Could not find module for " <> i
+
+-- |A module is a path dep if its import path starts with './' or '../'.
+isPathDep :: GoModule -> Bool
+-- I think this method of detection is crude, but I'm not sure how else to distinguish path deps from regular ones.
+-- If I don't assume that a path dep always starts with one of these prefixes, it's possible that what looks like an import path is actually a relative path to a local directory.
+-- One possible improvement is to only perform this check when a module is a replacement for another.
+-- Is this too unix specific?
+isPathDep GoModule{modulePath = ModulePath mP} = any (`isPrefixOf` mP) ["./", "../"]
```

Comment:
I agree this is a crude detection method, but I think this is valid and I do not think this is unix specific. I don't think this is unix specific because path deps are declared using `./` in the `go.mod` file and I would be surprised if that changed in a Windows environment. Either way, we have a higher tolerance for risk with path deps since we plan to fully support them soon.

Reply:
I was unsure when I wrote this, but was guessing that it was a requirement for local path deps to start with `./` or `../` because they'd be impossible to distinguish from regular module paths otherwise in the absence of a flag or something indicated it was a path dep.

I had another look here and it looks like this is the correct way to do it actually: https://go.dev/ref/mod#go-mod-file-replace